client/app/pages/dashboards/DashboardPage.jsx

```diff
@@ -164,6 +165,10 @@ function DashboardPage({ dashboardSlug, dashboardId, onError }) {
       })
       .catch(handleError);
   }, [dashboardId, dashboardSlug, handleError]);
+  
+  useEffect(() => {
+    addDashboardToRecentDashboards();
+  }, [addDashboardToRecentDashboards]);
```

Comment:
For sure that will work better, but this conflicts with what you have said about "make the same hook for both Query and Dashboard pages.", because in queries we need to have this callback so we can call it when the user execute the query, maybe its better to have only one hook but with the callback. What do you think?

Reply:
We can pass the "action" as a parameter to the Hook, or pass the service object instead of just the Id (Query or Dashboard) and "assume" the method will be the same from there.

Reply:
Why add a query to recents only when executing a query? I think it's reasonable to add it when viewing. If we see it "spams" the list too much, we can revise the logic.

Reply:
We also add the query to recents when viewing it! in the query view file:
![image](https://user-images.githubusercontent.com/55694628/92601176-e29f9180-f282-11ea-989f-91db553b5f1b.png)

Reply:
We also add the query to recents when viewing it! At the QueryView file: 
![image](https://user-images.githubusercontent.com/55694628/92601268-fd720600-f282-11ea-86a1-2ebda6d5791b.png)